src/Shared/Model/Mutators/BaseMutator.cs

```diff
@@ -0,0 +1,24 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CST.OpenSource.Model.Mutators
+{
+    /// <summary>
+    /// The base mutator to be implemented by other mutators.
+    /// </summary>
+    public abstract class BaseMutator
+    {
+        public string Mutator = "BASE_MUTATOR";
+
+        /// <summary>
+        /// Generates the typo squat mutations for a string.
+        /// </summary>
+        /// <param name="arg">The string to generate mutations for.</param>
+        /// <returns>A list of mutations with the name and reason.</returns>
+        public abstract IEnumerable<(string Name, string Reason)> Generate(string arg);
```

Comment:
What do you think about representing the output of this function as a custom `Mutation` object that contains this information (and possibly more in the future)? It should make working with this content a little bit easier and more concise with additional documentation in the class.

Reply:
I like this idea.